pkg/pipeline/pipeline.go

```diff
@@ -1384,34 +1377,78 @@ func fileHasSuffix(arr []string, str string) bool {
 	return false
 }
 
-func (b *Builder) CreateAssetFromFile(path string) (*Asset, error) {
+func (b *Builder) CreateAssetFromFile(filePath string, foundPipeline *Pipeline) (*Asset, error) {
 	isSeparateDefinitionFile := false
 	creator := b.commentTaskCreator
 
-	if fileHasSuffix(b.config.TasksFileSuffixes, path) {
+	if fileHasSuffix(b.config.TasksFileSuffixes, filePath) {
 		creator = b.yamlTaskCreator
 		isSeparateDefinitionFile = true
 	}
 
-	task, err := creator(path)
+	task, err := creator(filePath)
 	if err != nil {
 		if errors.As(err, &ParseError{}) {
 			return nil, err
 		}
 
-		return nil, errors.Wrapf(err, "error creating asset from file '%s'", path)
+		return nil, errors.Wrapf(err, "error creating asset from file '%s'", filePath)
 	}
 
 	if task == nil {
 		return nil, nil
 	}
 
-	task.DefinitionFile.Name = filepath.Base(path)
-	task.DefinitionFile.Path = path
+	// if the definition comes from a Python file the asset is always a Python asset, so force it
+	// at least that's the hypothesis for now
+	if strings.HasSuffix(task.ExecutableFile.Path, ".py") && task.Type == "" {
+		task.Type = AssetTypePython
+	}
+
+	task.DefinitionFile.Name = filepath.Base(filePath)
+	task.DefinitionFile.Path = filePath
 	task.DefinitionFile.Type = CommentTask
 	if isSeparateDefinitionFile {
 		task.DefinitionFile.Type = YamlTask
 	}
 
+	if foundPipeline != nil && foundPipeline.DefaultValues != nil {
+		defaultValuesBytes, err := json.Marshal(foundPipeline.DefaultValues)
```

Comment:
you can use https://github.com/go-viper/mapstructure/ for this instead of serializing/deserializing